quiche/src/dgram.rs

```diff
@@ -80,7 +82,23 @@ impl DatagramQueue {
     }
 
     pub fn pop(&mut self) -> Option<Vec<u8>> {
-        if let Some(d) = self.queue.as_mut().and_then(|q| q.pop_front()) {
+        let (data, clear) =
+            if let Some((urgency, q)) = self.queue.iter_mut().next() {
+                let data = q.pop_front();
+                let clear = if q.is_empty() {
+                    Some(*urgency)
+                } else {
+                    None
+                };
+                (data, clear)
+            } else {
+                (None, None)
+            };
+        
+        if let Some(urgency) = &clear {
+            self.queue.remove(urgency);
```

Comment:
what happens if the the BTreeMap has an entry for urgency 1, but the queue is empty? Should we move on to the next urgency? Seems like we should either loop to go to the next urgency (if any) or we should make sure this case never happens. This seems like a good scenario for a unit test